nservicebus/tools/migrate-to-native-delivery.md

```diff
@@ -0,0 +1,233 @@
+- `--prefix`: The prefix used for storage of timeouts. The default value is "TimeoutDatas"
+- `--ravenVersion`: The allowed values are "3.5" and "4"
+
+For SQL:
+
+- `--source`: The connection string to the database
+- `--dialect`: The SQL dialect used to access the database. Supported dialects: `MsSqlServer`
+
+### Preview
+
+To get a preview of endpoints and their status use the `preview` command with the following extra parameters.
+
+- `--target`: The connection string of the target transport used to validate destinations
+
+**RavenDB**
+
+```
+migrate-timeouts ravendb preview
+                        -t|--target <targetConnectionString>
+                        --serverUrl <serverUrl>
+                        --databaseName <databaseName>
+                        [--prefix] <prefix>
+                        [--ravenVersion] <ravenVersion>
+```
+
+**Sql Persistence**
+
+```
+migrate-timeouts sqlp preview
+                        -t|--target <targetConnectionString>
+                        --source <source>
+                        --dialect <sqlDialect>
+```
+
+NOTE: The listed endpoints will be the in the escaped form that is used to prefix the endpoints timeout table
+
+### Running a migration
+
+To run a migration for selected endpoint(s) use the `migrate` command with the following parameters.
+
+- `--target`: The connection string of the target transport
+- `--endpoint`(Optional): The endpoint to migrate.
+- `--allEndpoints`(Optional): Indicates to migrate all endpoints in one run
+- `--cutoffTime`(Optional): The time from which to start migrating timeouts, it makes sense to start migrating timeouts that will expire at least one day in the future. The format in which to specify the `cutoffTime` is `yyyy-MM-dd HH:mm:ss`. The migration tool will convert the specified `cutoffTime` to UTC time.
+
+NOTE: `--endpoint` and `--allEndpoints` arguments are mutually exclusive, either one is required.
+
+**RavenDB**
+
+```
+migrate-timeouts ravendb migrate
+                        -t|--target <targetConnectionString>
+                        --serverUrl <serverUrl>
+                        --databaseName <databaseName>
+                        [--prefix] <prefix>
+                        [--ravenVersion] <ravenVersion>
+                        [--endpoint] <endpointName>
+                        [-c|--cutofftime <cutofftime>]
+                        [--allendpoints]
+```
+
+**Sql Persistence**
+
+```
+migrate-timeouts sqlp migrate
+                        -t|--target <targetConnectionString>
+                        --source <source>
+                        --dialect <sqlDialect>
+                        [-c|--cutofftime <cutofftime>]
+                        [--endpoint] <endpointName>
+                        [--allendpoints]
+```
+
+NOTE: The listed endpoints will be the in the escaped form that is used to prefix the endpoints timeout table
+
+### Aborting a migration
+
+To abort an ongoing migration use the `abort` command.
```

Comment:
```suggestion
To abort an ongoing migration use the `abort` command. This will revert any pending timeouts that still need to be migrated.

NOTE: Any timeouts that have already been migrated will not be reverted.
```